flow/enhancers/socket_info.go

```diff
@@ -28,368 +30,310 @@ import (
-	inode string
+	tracer *tracer.Tracer
+	local  *cache.Cache
+	remote *cache.Cache
 }
 
-// Name return the SocketInfo enahancer name
+// Name returns the SocketInfo enhancer name
 func (s *SocketInfoEnhancer) Name() string {
 	return "SocketInfo"
 }
 
-func (s *SocketInfoEnhancer) resetScanProcNet() {
-	s.inodeNetMutex.Lock()
-	s.inodeNet = make(map[string]bool)
-	s.inodeNetMutex.Unlock()
-}
-
-func (s *SocketInfoEnhancer) needScanProcNet(procProcessRoot string) bool {
-	nsNet, err := os.Readlink(procProcessRoot + "/ns/net")
-	if err != nil {
-		return false
-	}
-
-	s.inodeNetMutex.Lock()
-	defer s.inodeNetMutex.Unlock()
-	if _, ok := s.inodeNet[nsNet]; !ok {
-		s.inodeNet[nsNet] = true
-		return true
-	}
-	return false
-}
-
-func (s *SocketInfoEnhancer) parseNetUDPTCP(c *socketCache, path string) {
-	u, err := os.Open(path)
-	if err != nil {
-		return
+func ipToHex(ip46 net.IP) (s string) {
+	ip := ip46.To4()
+	if ip == nil {
+		ip = ip46
 	}
-	defer u.Close()
 
-	r := bufio.NewReader(u)
-	for {
-		line, err := r.ReadString('\n')
-		if err != nil {
-			return
-		}
-		i := strings.Index(line, ":")
-		if i < 0 {
-			continue
-		}
-		var localIPPort, remoteIPPort string
-		var state, wq, rq int
-		var timer, timeout, retrs, uid, probes int
-		var ino uint64
-		var refcnt, sk, rto, ato, qack int
-		var cwnd, ssthresh, opt int
-
-		fmt.Sscanf(line[i+1:], "%s %s %x %x:%x %x:%x %x %d %d %d %d %llx %d %d %d %d %d %[^\n]\n",
-			&localIPPort, &remoteIPPort,
-			&state, &wq, &rq,
-			&timer, &timeout, &retrs, &uid, &probes, &ino,
-			&refcnt, &sk, &rto, &ato, &qack,
-			&cwnd, &ssthresh, &opt)
-
-		inode := strconv.FormatUint(ino, 10)
-		_, ok := c.inode.Get(inode)
-		if ok {
-			continue
-		}
-		si := SocketInfo{
-			Uid:          uid,
-			LocalIPPort:  localIPPort,
-			RemoteIPPort: remoteIPPort,
-		}
-		c.inode.Set(inode, si, cache.DefaultExpiration)
-		c.localAddr.Set(localIPPort, cacheAddr{inode: inode}, cache.DefaultExpiration)
-		c.localRemoteAddr.Set(localIPPort+remoteIPPort, cacheAddr{inode: inode}, cache.DefaultExpiration)
+	for i := range ip {
+		s += fmt.Sprintf("%02X", ip[(len(ip)-1)-i])
 	}
+	return
 }
 
-func (s *SocketInfoEnhancer) updateProcNet(procProcessRoot string) {
-	/* scan only new one based on network namespace */
-	if s.needScanProcNet(procProcessRoot) == false {
-		return
-	}
-	s.parseNetUDPTCP(s.cacheUDP, procProcessRoot+"/net/udp")
-	s.parseNetUDPTCP(s.cacheUDP, procProcessRoot+"/net/udp6")
-	s.parseNetUDPTCP(s.cacheTCP, procProcessRoot+"/net/tcp")
-	s.parseNetUDPTCP(s.cacheTCP, procProcessRoot+"/net/tcp6")
+func hashIPPort(ip net.IP, port uint16) string {
+	return fmt.Sprintf("%s:%04X", ipToHex(ip), port)
 }
 
-var rProcPidTid = regexp.MustCompile("/proc/(\\d+)/task/(\\d+)$")
-
-func parseProcStat(procProcessRoot string, si *SocketInfo) {
-	// Retreive binary name
-	exe, err := os.Readlink(procProcessRoot + "/exe")
+func getProcessInfo(pid int) (*flow.SocketInfo, error) {
+	processPath := fmt.Sprintf("/proc/%d", pid)
```

Comment:
we losing the threadID (tid) here, a task could have a different name than the pid
would be possible to retrieve the info from tcptrace